### ISSUE TITLE
plpgsql-keyword-case: 'row_count' variable name flagged as a keyword (false positive)

### ISSUE THREAD

**Issue by baseballyama** [baseballyama/eslint-plugin-postgresql#188]
## Bug

The `plpgsql-keyword-case` rule incorrectly flags `row_count` declared as a PL/pgSQL **variable** as if it were the built-in `ROW_COUNT` keyword used in `GET DIAGNOSTICS`.

`ROW_COUNT` is only a keyword in the context of `GET DIAGNOSTICS xxx = ROW_COUNT`. Using `row_count` as a regular identifier (variable name in `DECLARE`, target of `SELECT INTO`, comparison in `IF`) is valid lower-case SQL and should not be flagged.

## Plugin version

`eslint-plugin-postgresql@0.18.0`

## Reproduction

```sql
CREATE FUNCTION example() RETURNS TRIGGER
LANGUAGE PLPGSQL
AS $$
DECLARE
  row_count INTEGER;
BEGIN
  SELECT COUNT(*) INTO row_count FROM some_table;
  IF row_count > 0 THEN
    RAISE NOTICE 'rows: %', row_count;
  END IF;
  RETURN NEW;
END;
$$;
```

This produces three false positives:

```
error  PL/pgSQL keyword 'row_count' should be uppercase: 'ROW_COUNT'   postgresql/plpgsql-keyword-case
```

at the `DECLARE`, the `INTO`, and the `IF` lines.

## Expected behavior

The rule should only flag `row_count` (or other diagnostic item names like `pg_context`, `returned_sqlstate`, `column_name`, `constraint_name`, `pg_datatype_name`, `message_text`, `table_name`, `schema_name`, `pg_exception_detail`, `pg_exception_hint`, `pg_exception_context`) when they appear on the **right-hand side of `GET DIAGNOSTICS ... = `** (or `GET STACKED DIAGNOSTICS ... = `). Anywhere else they are normal identifiers.

## Why this matters

`row_count` is a natural variable name for "the number of rows matching some condition". A common pattern in trigger functions is to `SELECT COUNT(*) INTO row_count FROM ...` and then branch on it. Renaming the variable to avoid the rule is unnecessary friction.

## Related rule

The same kind of false-positive likely exists for other GET DIAGNOSTICS item names if they are used as identifiers, but the most common one in practice is `row_count`.
